jupyterhub/tests/mocking.py

```diff
@@ -74,17 +74,30 @@ def stop(self):
         yield super().stop()
 
 
+def fake_gen(fun):
+    fun.__tornado_coroutine__ = True
+    return fun
+
 class NeverSpawner(MockSpawner):
     """A spawner that will never start"""
-    
+
     @default('start_timeout')
     def _start_timeout_default(self):
         return 1
-    
+
+    @fake_gen
```

Comment:
returning a Future should be valid. If it's not, the check is making an inappropriate assertion.